File: src/main/java/com/example/sneakerx/dtos/shop/UpdateShopRequest.java
package com.example.sneakerx.dtos.shop;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UpdateShopRequest {
    private Integer shopId;
    private String shopName;
    private String shopDescription;
    private String shopLogo;
}

File: lib/src/modules/seller_info/dtos/update_shop_request.dart
class UpdateShopRequest {
  final int shopId;
  final String shopName;
  final String shopDescription;
  final String? shopLogo;

  UpdateShopRequest({
    required this.shopId,
    required this.shopName,
    required this.shopDescription,
    this.shopLogo,
  });

  Map<String, dynamic> toJson() {
    return {
      'shopId': shopId,
      'shopName': shopName,
      'shopDescription': shopDescription,
      'shopLogo': shopLogo
    };
  }
}